src/controllers/EthereumBlockchainController.ts

```diff
@@ -387,6 +437,32 @@ export class EthereumBlockchainController extends AbstractBlockchainController {
 		throw new Error('Method not implemented.');
 	}
 
+	async getTokenAttachments(msg: IEVMMessage): Promise<YlideTokenAttachment> {
+		const decodedMsgId = decodeEvmMsgId(msg.msgId);
+		const mailer = this.mailers.find(m => m.link.id === decodedMsgId.contractId);
+		if (!mailer) {
+			throw new Error('This message does not belongs to this blockchain controller');
+		}
+		if (mailer.wrapper instanceof EthereumMailerV9Wrapper) {
+			const tokenAttachmentAddress = msg.$$meta.tokenAttachment;
+			if (!tokenAttachmentAddress || tokenAttachmentAddress === ethers.constants.AddressZero) {
+				throw new Error('Message has no token attachment');
+			}
+			const tokenAttachmentLink = EVM_CONTRACTS[this.network].payContracts?.find(
+				c => c.address === tokenAttachmentAddress,
+			);
+			if (!tokenAttachmentLink) {
+				throw new Error('Message has no token attachment');
+			}
+			const attachments = await mailer.wrapper.mailing.getTokenAttachments(tokenAttachmentLink, msg);
+			return {
+				kind: TokenAttachmentContractType.Pay,
+				attachments,
+			};
+		}
+		throw new Error('Method not implemented.');
```

Comment:
Here is better to return `null` or empty array. It is not responsibility of caller to check if message is V9 or less and it seems to be too much to throw error in this case